src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackValidations.java

```diff
@@ -112,6 +112,14 @@ private CallbackValidations() {
             : invalid("Exception case has no Id");
     }
 
+    @Nonnull
+    public static Validation<String, CaseDetails> hasId(CaseDetails theCase) {
+        return theCase != null
+            && theCase.getId() != null
+            ? valid(theCase)
+            : invalid("Exception case has no Id");
+    }
```

Comment:
As far as I can tell ALL other `hasX` methods in this class return X.
`hasAnId` already exists and returns an Id.
This looks like some kind of workaround.

Can you, instead of adding this new method, call existing `hasAnId` method and then pass `CaseDetails` wherever you need to?